Comments/components/commentMorph.js

```diff
@@ -26,42 +25,67 @@ export class CommentGroupMorph extends Morph {
     this.ui.groupNameLabel.textString = this.referenceMorph.name;
   }
 
-  async getCommentMorphs (comments) {
-    const commentMorphs = [];
-    await Promise.all(comments.map(async (comment) => {
-      const commentMorph = await resource('part://CommentComponents/comment morph master').read();
-      this.commentIndicators.push(CommentIndicator.for(this.referenceMorph, comment));
-      commentMorph.initialize(comment, this.referenceMorph);
-      commentMorphs.push(commentMorph);
-    }));
-    this.ui.commentCountLabel.textString = commentMorphs.length;
-    return commentMorphs;
+  async addCommentMorph (comment) {
+    const commentMorph = await resource('part://CommentComponents/comment morph master').read();
+    const commentIndicator = CommentIndicator.for(this.referenceMorph, comment);
```

Comment:
Warum verwaltet die Gruppe den CommentIndicator eines Comments?

Reply:
It would make more sense architecturally, but it doesn't work. Since we already invested about 6 person hours into this, we will not change it at this time. The comment indicators just don't do what we expect, possibly because of async.